integrations/appboy/test/onlyTrackKnownUsersOnWeb.test.js

```diff
@@ -0,0 +1,261 @@
+var Analytics = require('@segment/analytics.js-core').constructor;
+// var integration = require('@segment/analytics.js-integration');
+var sandbox = require('@segment/clear-env');
+var tester = require('@segment/analytics.js-integration-tester');
+var Appboy = require('../lib/');
+var assert = require('assert');
+var sinon = require('sinon');
+
+describe('Appboy with onlyTrackKnownUsersOnWeb enabled', function() {
+  var analytics;
+  var appboy;
+  var options = {
+    apiKey: '7c664901-d8c0-4f82-80bf-e7e7a24478e8',
+    automaticallyDisplayMessages: false,
+    safariWebsitePushId: '',
+    enableHtmlInAppMessages: false,
+    trackAllPages: false,
+    trackNamedPages: false,
+    customEndpoint: '',
+    version: 1,
+    onlyTrackKnownUsersOnWeb: true, // Default off.
```

Comment:
These tests have the setting enabled. I picked a subset of the tests from `index.test.js` , reduced the validations, and tested the new behavior.